File: api/admin/category/category.go
package category

import (
	"gitee.com/jieepre/go-site/internal/model"
	"gitee.com/jieepre/go-site/internal/pkg/core"
	"gitee.com/jieepre/go-site/pkg/result"
	"github.com/gin-gonic/gin"
	"strconv"
)

type Handler struct {
	*core.Context
}

func (h *Handler) SaveCategory(c *gin.Context) {
	category := model.Category{}
	err := c.ShouldBindJSON(&category)
	if err != nil {
		result.Error(c, err.Error())
		return
	}
	err = h.Service.CategoryService.Save(category)
	if err != nil {
		result.Error(c, err.Error())
		return
	}
	result.Ok(c, nil)
}
func (h *Handler) UpdateCategory(c *gin.Context) {
	category := model.Category{}
	err := c.ShouldBindJSON(&category)
	if err != nil {
		result.Error(c, err.Error())
		return
	}
	err = h.Service.CategoryService.Update(category)
	if err != nil {
		result.Error(c, err.Error())
		return
	}
	result.Ok(c, nil)
}
func (h *Handler) UpdateCategoryState(c *gin.Context) {
	category := model.Category{}
	err := c.ShouldBindJSON(&category)
	if err != nil {
		result.Error(c, err.Error())
		return
	}
	err = h.Service.CategoryService.Update(category)
	if err != nil {
		result.Error(c, err.Error())
		return
	}
	result.Ok(c, nil)
}
func (h *Handler) DeleteCategory(c *gin.Context) {
	param := c.Param("categoryId")
	categoryId, _ := strconv.ParseInt(param, 0, 64)

	posts, err := h.Service.PostService.GetPostsByCategoryId(categoryId)
	if len(posts) > 0 {
		result.Error(c, "不能删除该栏目")
		return
	}
	err = h.Service.CategoryService.Delete(int(categoryId))
	if err != nil {
		result.Error(c, err.Error())
		return
	}
	result.Ok(c, nil)
}
func (h *Handler) DetailCategory(c *gin.Context) {
	param := c.Param("categoryId")
	categoryId, _ := strconv.ParseInt(param, 0, 64)
	categoryInfo, err := h.Service.CategoryService.GetCategory(uint32(categoryId))
	if err != nil {
		result.Error(c, err.Error())
		return
	}
	result.Ok(c, categoryInfo)
}
func (h *Handler) GetCategoryList(c *gin.Context) {
	categories, err := h.Service.CategoryService.GetCategoryList()
	if err != nil {
		result.Error(c, err.Error())
		return
	}
	result.Ok(c, categories)
}

File: frontend/src/api/category.ts
import { http } from "@/utils/http";

type Result = {
  code: number;
  message: string;
  payload?: any;
};

/** 获取分类管理列表 */
export const getCategoryList = () => {
  return http.request<Result>("get", "/api/v1/site/category/list");
};
/**更新分类**/
export const updateCategory = (data?: object) => {
  return http.request<Result>("put", "/api/v1/site/category/update", {
    data
  });
};
/**更新分类**/
export const addCategory = (data?: object) => {
  return http.request<Result>("post", "/api/v1/site/category/save", {
    data
  });
};

/**删除分类**/
export const deleteCategory = (categoryId?: number) => {
  return http.request<Result>(
    "delete",
    "/api/v1/site/category/delete/" + categoryId
  );
};

/**更新分类状态**/
export const changeCategoryState = (data?: object) => {
  return http.request<Result>("put", "/api/v1/site/category/update-state", {
    data
  });
};
/**获取分类**/
export const getCategory = (categoryId?: number) => {
  return http.request<Result>(
    "get",
    "/api/v1/site/category/detail/" + categoryId
  );
};
